### ISSUE TITLE
CSV STRICT_CHECK_FOR_QUOTING does not enclose value with newline character in double quotes

### ISSUE THREAD

**Issue by zhuhw** [FasterXML/jackson-dataformats-text#479]
If I enable `STRICT_CHECK_FOR_QUOTING` 
```
public static void main(String[] args) throws JsonProcessingException {
    CsvFactory csvFactory = new CsvFactory();
    csvFactory.enable(CsvGenerator.Feature.STRICT_CHECK_FOR_QUOTING);

    ObjectMapper objectMapper = Jackson.newObjectMapper(csvFactory);
    ObjectWriter writer = objectMapper.writer();
    String s = writer.writeValueAsString("line1\nline2");
    System.out.println(s);
}
```
The output is:
```
line1
line2
```
But when I disable the Feature, output becomes 
```
"line1
line2"
```
If I change the newline character to a comma, the value is enclosed no matter I enable or disable this Feature:
```
"line1,line2"
```

This does not follow the https://www.ietf.org/rfc/rfc4180.txt item 2.6

**Comment by cowtowncoder** [comment]
That does seem wrong to me, thank you for reporting.

Quick question: which version did you test with?

**Comment by cowtowncoder** [comment]
@zhuhw That is a rather old version (almost 5 years ago) and a lot has changed since then. Does the problem still occur with, say, 2.17.1?

I can try it now, but for future reference it is very strongly recommended to test with the latest release version to avoid wasting everyone's time on bugs that have already been fixed (or behavior changed).

**Comment by cowtowncoder** [comment]
For example, #220 was fixed for 2.11.3 and seems like that might be fix for your issue @zhuhw ?

**Comment by cowtowncoder** [comment]
Assuming this was fixed since I have seen no updates; closing.

May be re-opened with reproduction against recent version, if this still occurs.

**Comment by the8thC** [comment]
@cowtowncoder I found this issue during migration from 2.8.8 to 2.20.1 and it seems to reproduce with all versions from 2.9.+ to 2.20.+

```kotlin
    @Test
    fun `jackson dataformats csv - issue 479`() {
        val objectMapper: ObjectMapper = CsvMapper().apply {
            registerModule(SimpleModule())
            enable(CsvGenerator.Feature.STRICT_CHECK_FOR_QUOTING)
        }
        val csvSchema = CsvSchema.builder().apply {
            addColumn("c1")
            addColumn("c2")
            setUseHeader(true)
            setLineSeparator("\r\n")
        }.build()

        val outputStream = ByteArrayOutputStream()
        val writer = objectMapper
            .writer(csvSchema)
            .writeValues(
                OutputStreamWriter(outputStream, Charset.forName("UTF-8"))
            )

        val values = mapOf("c1" to "line\nbroken", "c2" to "line")
        writer.write(values)

        val result: String = outputStream.toString(StandardCharsets.UTF_8)
        println("printing:\n${result}")
        assertThat(result).contains("\"line\nbroken\"")
    }
```


2.9.+ and 2.20.1 output:
```
printing:
c1,c2
line
broken,line



Expecting actual:
  "c1,c2
line
broken,line
"
to contain:
  ""line
broken""
```

2.8.8 output:
```
printing:
c1,c2
"line
broken",line
```


And if I remove `enable(CsvGenerator.Feature.STRICT_CHECK_FOR_QUOTING)`:

2.20.1 and 2.8.8 have the same output:
```
printing:
c1,c2
"line
broken",line
```

Same also happens if the line separator used is `setLineSeparator("\n")`

**Comment by cowtowncoder** [comment]
Fixed: will be in 3.1.0 and backported for 2.21.0 as well.
